newsletter_laravel/app/Services/NewsletterService.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace App\Services;
+
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Client;
+use Exception;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class NewsletterService {
+
+    private Client $client;
+
+    private SymfonyStyle $io;
+
+    public function __construct()
+    {
+        //Ne dela ce imam /services/v5 tukaj namesto v requestu?
+        $this->client = new Client(['base_uri' => 'https://www.pivotaltracker.com']);
+    }
+
+    /**
+     * Fetches all stories with the state set to accepted
+     *
+     * @param SymfonyStyle $io
+     * @return array|void|null
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function exportStories(SymfonyStyle $io)
+    {
+        $this->io = $io;
+        try {
+            $response = $this->client->request('GET', '/services/v5/projects/' . env('PIVOTAL_PROJECT_ID') . '/stories', [
+                'headers' => [
+                    'X-TrackerToken' => env('PIVOTAL_API_TOKEN'),
+                ],
+                'query' => [
+                    'with_state' => 'accepted',
+                ]
+            ]);
+
+            $data = $this->filterByReviews(json_decode($response->getBody(), true));
+            return $data;
+
+        } catch (RequestException $e) {
+            $this->io->error($e->getMessage());
+        }
+    }
+
+    /**
+     * Filters the given stories so that they only include those with a certain review type set to pass
+     *
+     * @param $data
+     * @return array|void
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function filterByReviews($data)
+    {
+        try {
+            $output = [];
+            foreach($data as $story) {
+                $response = $this->client->request('GET', '/services/v5/projects/' . env('PIVOTAL_PROJECT_ID') . '/stories/' . $story['id'] . '/reviews', [
+                    'headers' => [
+                        'X-TrackerToken' => env('PIVOTAL_API_TOKEN'),
```

Comment:
It's bad practice to use here `env` method. Implement config file like `config/pivotal-tracker.php` and then call here `config('pivotal-tracker.api_token')`